src/mbgl/renderer/image_atlas.hpp

```diff
@@ -51,8 +59,12 @@ class ImageAtlas {
     PremultipliedImage image;
     ImagePositions iconPositions;
     ImagePositions patternPositions;
+
+    void patchUpdatedImages(gfx::Context&, gfx::Texture&, const ImageManager&);
+private:
+    void patchUpdatedImage(gfx::Context&, gfx::Texture&, ImagePosition& position, const ImageManager& imageManager, const std::string& name, uint16_t version);
 };
 
-ImageAtlas makeImageAtlas(const ImageMap&, const ImageMap&);
+ImageAtlas makeImageAtlas(const ImageMap&, const ImageMap&, const std::unordered_map<std::string, uint32_t>& versionMap);
```

Comment:
As we're accumulating more objects that work together to form an ImageAtlas, maybe it's time to bundle them up together in a follow up PR?

Reply:
Sounds like a good idea